src/lib/components/AssistantSettings.svelte

```diff
@@ -390,12 +397,25 @@
 
 		<div class="col-span-1 flex h-full flex-col">
 			<span class="mb-1 text-sm font-semibold"> Instructions (system prompt) </span>
-			<textarea
-				name="preprompt"
-				class="mb-20 min-h-[8lh] flex-1 rounded-lg border-2 border-gray-200 bg-gray-100 p-2 text-sm"
-				placeholder="You'll act as..."
-				value={assistant?.preprompt ?? ""}
-			/>
+			<div class="relative mb-20 flex min-h-[8lh] flex-1 grow flex-col">
+				<textarea
+					name="preprompt"
+					class="flex-1 rounded-lg border-2 border-gray-200 bg-gray-100 p-2 text-sm"
+					placeholder="You'll act as..."
+					bind:value={systemPrompt}
+				/>
+				{#if modelId}
+					{@const model = models.find((_model) => _model.id === modelId)}
+					{#if model?.tokenizer && systemPrompt}
+						<TokensCounter
+							classNames="absolute bottom-2 right-2"
+							prompt={systemPrompt}
+							modelTokenizer={model.tokenizer}
+							max_new_tokens={model?.parameters?.max_new_tokens}
```

Comment:
use `truncate` value instead?

Reply:
handled in f47c6bba93b8e8c8416a586138d37921631cb222